docs/hibernating-clusters.md

```diff
@@ -0,0 +1,146 @@
+The actuator, given a ClusterDeployment's InfraID selects machines using a method appropriate to the cloud provider (tags/name prefix/resource group).
+
+Option 2:
+The hibernation controller uses the machine API on the target cluster to determine which machines belong to the cluster. It then stores the machine IDs
+in the clusterdeployment (or a separate CR), then uses those machine IDs to start the cluster again.
+
+#### Hibernation Actuator
+The hibernation controller relies on an actuator to work with cloud provider machines.
+This is the interface for the actuator:
+
+```go
+type HibernationActuator interface {
+  // StopMachines will start machines belonging to the given ClusterDeployment
+  StopMachines(cd *hivev1.ClusterDeployment) error
+
+  // StartMachines will select machines belonging to the given ClusterDeployment
+  StartMachines(cd *hivev1.ClusterDeployment) error
+
+  // MachinesRunning will return true if the machines associated with the given
+  // ClusterDeployment are in a running state.
+  MachinesRunning(cd *hivev1.ClusterDeployemnt) (bool, error)
+
+  // MachinesStopped will return true if the machines associated with the given
+  MachinesStopped(cd *hivev1.ClusterDeployment) (bool, error)
+}
+```
+
+#### Handling Incompatible OpenShift Versions and Cloud Provider
+OpenShift versions earlier than 4.4.8 do not support stopping and starting a cluster without additional work
+to restore etcd. In the case that the cluster deployment's `status.clusterVersionStatus.desired.version` is
+less than 4.4.8, the hibernation controller will set the Hibernating condition to `false` and set the reason
+to Unsupported. This will also be the case if the cluster's cloud provider is not currently supported.
+When later reconciling cluster deployments with this condition set, the hibernation controller will
+continue to check the version in case the cluster is upgraded and eventually is able to be hibernated.
+
+#### Approving CSRs
+In the case that CSRs must be approved for a cluster that has had its certificates expired while hibernating,
+we should follow similar checks as the [cluster machine approver](https://github.com/openshift/cluster-machine-approver/blob/0f50c7bfe9b309ce01937274598f5a807d9545df/csr_check.go)
+to ensure we are not introducing an additional security exposure.
+
+#### Resuming from a Hibernating State
+When a cluster is hibernated, the unreachable controller should properly set the unreachable condition on
+the cluster once it stops responding. This will cause other controllers like the remotemachineset controller to
+stop trying to reconcile the cluster. However, when having resumed a cluster, the hibernation controller should
+remove the unreachable condition if set so that other controllers can start processing the cluster as well. The
+reason is that the unreachable controller could take up to 2 hours to remove the unreachable condition from the
+cluster deployment on its own.
```

Comment:
I think unreachable controller has a connectivity recheck delay of 2 hours when cluster is reachable. When cluster is unreachable we do connectivity recheck more frequently. So may be hibernation controller need not delete the unreachable condition, unreachable controller will do it.

Reply:
Thanks, now updated to indicate that the unreachable controller should handle everything.